keycloak-create-test-users/src/main/java/CsvUserImport.java

```diff
@@ -0,0 +1,130 @@
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        String line = reader.readLine(); //ignore first line
+        line = reader.readLine(); 
+        List<User> users = new ArrayList<>();
+        while (line!=null){
+            String[] parts = line.split(",");
+            if (parts.length>=8){
+                User user = parseUserFromLine(parts, "PLR");
+                users.add(user);
+                System.out.println("Parsed User:"+(user.getUsername()));
+            }else{
+                System.out.println("Can't parse: "+line);
+            }
+            line = reader.readLine();
+        }
+        
+        if (args != null && args.length != 0) {
+            configPath = args[0];
+        } else {
+            URL defaultLocation = Main.class.getClassLoader().getResource("configuration.properties");
+            configPath = new File(defaultLocation.toURI()).getAbsolutePath();
+        }
+        LOG.info(String.format("Configuration file expected at '%s'.", configPath));
+
+        // initializes the userService
+        userService = new UserService(configPath);
+        for (User user: users){
+            try{
+                if (userService.userExists(user)){
+                    System.out.println("User "+user.getUsername()+" exists");
+                }else{
+                    System.out.println("User "+user.getUsername()+" does not exist... to be created");
+                    userService.createUserInKeycloak(user.getUserRepresentation());
+                }
+                userService.addClientRolesInKeyCloak(user);
+            }catch(Exception e){
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+    
+    /**
+     * Creates a User object from a line in the CSV
+     * @param line String[] - parsed line of CSV file
+     * @param client String - Name of KeyCloak client to assign role to
+     * @return User
+     */
+    public static User parseUserFromLine(String[] line, String client){
+        String[] names = line[1].split(" ");
+        String email = line[3].toLowerCase();
+        String domain = line[5].toLowerCase();
+        String role = line[2];
+        String userName = line[7].toLowerCase();
+        User user = new User(userName);
+        user.getUserRepresentation().setFirstName(names[0]);
+        user.getUserRepresentation().setLastName(names[1]);
+        user.getUserRepresentation().setEmail(email);
+        user.recordClientRoles(client, role);
+        String org = line[0].substring(0,line[0].indexOf("-"));
+        user.addAttribute("org_details", getOrgMapping(org));
+       return user;
+    }
+    
+    /**
+     * Maps an organization id to the appropriate keycloak org attribute
+     * @param org - String
+     * @return String
+     */
+    public static String getOrgMapping(String org){
+        if (orgMap==null){
+            //Lazy init map
+            orgMap = new HashMap<>();
```

Comment:
We could just pull the orgs from UMC Prod to be used here -> https://user-management.hlth.gov.bc.ca/organizations.json
...that being said, it might not live there much longer.